src/workerd/api/http.c++

```diff
@@ -417,6 +418,143 @@ bool Headers::inspectImmutable() {
   return guard != Guard::NONE;
 }
 
+// -----------------------------------------------------------------------------
+// serialization of headers
+//
+// http-over-capnp.capnp has a nice list of common header names, taken from the HTTP/2 standard.
+// We'll use it as an optimization.
+//
+// Note that using numeric IDs for headers implies we lose the original capitalization. However,
+// the JS Headers API doesn't actually give the application any way to observe the capitalization
+// of header names -- it only becomes relevant when serializing over HTTP/1.1. And at that point,
+// we are actually free to change the capitalization anyway, and we commonly do (KJ itself will
+// normalize capitalization of all registered headers, and http-over-capnp also loses
+// capitalization). So, it's certainly not worth it to try to keep the original capitalization
+// across serialization.
+
+// If any more headers are added to the CommonHeaderName enum later, we should be careful about
+// introducing them into serialization. We need to roll out a change that recognizes the new IDs
+// before rolling out a change that sends them. MAX_COMMON_HEADER_ID is the max value we're willing
+// to send.
+static constexpr uint MAX_COMMON_HEADER_ID =
+    static_cast<uint>(capnp::CommonHeaderName::WWW_AUTHENTICATE);
+
+static kj::Array<kj::StringPtr> makeCommonHeaderList() {
```

Comment:
non-blocking nit: `api/http.c++~ is already fairly large... I'd prefer if these kinds of utility methods could be moved out to a separate `util/http.h` or something but obviously not critical.

Reply:
I kind of prefer to keep the code next to where it is used. That said, it would make a lot of sense (in a separate PR) to break up http.h/c++ into separate files for Headers, Request, Response, etc.